designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala

```diff
@@ -34,7 +35,11 @@ class DefinitionResources(modelDataProvider: ProcessingTypeDataProvider[ModelDat
     } ~ path("processDefinitionData" / "services") {
       get {
         complete {
-          modelDataProvider.mapValues(_.processDefinition.services.mapValuesNow(UIProcessObjectsFactory.createUIObjectDefinition(_, processCategoryService))).all
+          modelDataProvider.mapValues(
+            _.processDefinition.services.filter {
+              case (_, definition) => definition.categories.forall(_.exists(user.can(_, Read)))
```

Comment:
Why for all? One isn't enough to run the service?

Reply:
Ane where are the tests?! :)

Reply:
`forall` does not work like loop "for all elements". In this case, on `Option[List[String]]` type, it returns `true` if Option is `None` - otherwise it goes till it meets first `true` in given predicate. Take a look on REPL example

```
@ val optList = Option(List(1,2,3,4,5)).forall(_.exists(a => {println(a); a>1})) 
1
2
optList: Boolean = true
```

Reply:
Ahh I thought in categories is List[String] not Option[List]] ;) It's allright! But still filtering should be done earlier :)

Reply:
IMO if category is not set for service then we should return it - if somebody wants to hide some component, then this component should contain category for possibility to set correct access rights.

Reply:
And I think that you thought same way, since you've done exactly same assumptions :) 
```
def availableForCategory(category: String): Boolean = categories.isEmpty || categories.exists(_.contains(category))
```

Reply:
I've also added test case